### ISSUE TITLE
Plugin assumes that bash is present in container

### ISSUE THREAD

**Issue by gh2k** [buildkite-plugins/docker-buildkite-plugin#23]
The command is run like this:
```
docker run "${args[@]}" "${BUILDKITE_PLUGIN_DOCKER_IMAGE}" bash -c "${BUILDKITE_COMMAND}"
```

When using Windows Server containers this doesn't work without some messing around. In my environment `bash -c` could be safely dropped. I would have expected that when adding the buildkite command to my pipeline file it would be the entire cmd section, rather than wrapped.

bash also isn't present in some minimal Linux containers IIRC (I think Alpine just has 'sh', or at least it used to)

Personally I'd prefer this to drop the `bash -c` entirely and assume that the pipeline author would add this if required. If there's a reason for this being there that I'm missing, perhaps you could provide a mechanism for overriding the interpreter in the config. For example:
```
  - command:  build-windows.ps1
    name: Windows build
    plugins:
      docker:
        image: microsoft/windowsservercore:1709
        always-pull: true
        interpreter: powershell -Command
    agents:
      windows: true
```

**Comment by lox** [comment]
This is a really tricky one. See https://github.com/buildkite-plugins/docker-compose-buildkite-plugin/pull/124 for some more context.

**Comment by toolmantim** [comment]
We could potentially also skip an interpreter for simple commands?

**Comment by lox** [comment]
>We could potentially also skip an interpreter for simple commands?

@toolmantim yeah, that is a possibility. Perhaps we should just error for multi-line commands when there isn't a `shell` set and don't do it automatically for folks.

**Comment by toolmantim** [comment]
Fixed in #50 @gh2k. I'll put out a release now. But we're going to do a major version bump to remove the default shell behaviour, and have the shell an opt-in thing, with some detections if people are trying to run bash-isms without having shell enabled.
